### ISSUE TITLE
cd does not work if a directory starts with the character -

### ISSUE THREAD

**Issue by bugnano** [fish-shell/fish-shell#6071]
I'm using fish version 2.7.1 running on Fedora 29 x86_64.
Let's say for example that I create a directory like this:

```
mkdir -- -mydirectory
```

this creates a directory called -mydirectory (note that its name begins with the `-` character)

now I want to go to that directory.

If I try to run:

```
cd -mydirectory
```

it fails with:

```
cd: Unknown option '-mydirectory/'
/usr/share/fish/functions/cd.fish (line 30): 
    builtin cd $argv
    ^
in function 'cd'
	called on standard input
	with parameter list '-mydirectory/'
```

ok, let's try with:

```
cd -- -mydirectory
```

and it fails with the error:

```
Too many args for cd command
```

So I don't know any other method to cd to that directory

**Comment by zanchey** [comment]
You can also use `cd ./-mydirectory`, which makes `nextd` and `prevd` work.

Probably the `cd` function should be converted to argparse.

**Comment by zanchey** [comment]
Fixed in 2e6ab0b418f2fb17fcbf18ef5b118bc342a1118a.
